swap/src/asb/kraken.rs

```diff
@@ -77,8 +77,11 @@ impl RateService {
             while let Some(msg) = rate_stream.next().await {
                 let msg = match msg {
                     Ok(Message::Text(msg)) => msg,
+                    Ok(Message::Close(..)) => {
+                        let _ = rate_update.send(Err(Error::CloseMessage));
```

Comment:
I think we should log an error here and also investigate the `CloseFrame` if it has useful information.